pkg/codegen/docs/gen_creation_example_for_csharp.go

```diff
@@ -0,0 +1,258 @@
+package docs
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/dotnet"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+	"strings"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+)
+
+func csharpNamespaceName(namespaces map[string]string, name string) string {
+	if ns, ok := namespaces[name]; ok {
+		return ns
+	}
+
+	// name could be a qualified module name so first split on /
+	parts := strings.Split(name, tokens.QNameDelimiter)
+	for i, part := range parts {
+		names := strings.Split(part, "-")
+		for j, name := range names {
+			names[j] = title(name, "csharp")
+		}
+		parts[i] = strings.Join(names, "")
+	}
+	return strings.Join(parts, ".")
+}
+
+func csharpPropertyNameOverrides(properties []*schema.Property) map[string]string {
+	overrides := make(map[string]string)
+	for _, property := range properties {
+		foundOverride := false
+		if csharp, ok := property.Language["csharp"]; ok {
+			if options, ok := csharp.(dotnet.CSharpPropertyInfo); ok {
+				overrides[property.Name] = options.Name
+				foundOverride = true
+			}
+		}
+
+		if !foundOverride {
+			overrides[property.Name] = property.Name
+		}
+	}
+
+	return overrides
+}
+
+func resolveCSharpPropertyName(property string, overrides map[string]string) string {
+	foundOverride, ok := overrides[property]
+	if ok {
+		return title(foundOverride, "csharp")
+	}
+
+	return title(property, "csharp")
+}
+
+func genCreationExampleSyntaxCSharp(r *schema.Resource) string {
+	pkgDef, _ := r.PackageReference.Definition()
```

Comment:
I think panic'ing here is good because this shouldn't really work with `schema.Resource` that hasn't been initialized from. a bound schema. Will add that here ✅